src/components/TaskItem/Drawer/Drawer.tsx

```diff
@@ -58,86 +65,131 @@ export function TaskItemDrawer({
         enqueueSnackbar('Faith step updated successfully!', {
           variant: 'success'
         })
+        onDialogClose()
       } catch {
         enqueueSnackbar('Failed to update faith step. Please try again!', {
-          variant: 'error'
+          variant: 'error',
+          persist: true
         })
       }
     }
   }
+
+  function onDialogClose(): void {
+    setPostType(undefined)
+  }
+
+  function onClick(status: Task['status']): () => void {
+    return function (): void {
+      setTaskStatus(status)
+      if (status !== 'incomplete') {
+        setPostType(getPostType(status))
+      } else {
+        onSubmit('incomplete')()
+      }
```

Comment:
It's possible that the task.post* can be empty. It is not always a requirement that those fields have a value.

```suggestion
      const postType = getPostType(status)
      setPostType(postType)
      if (postType == null) {
        onSubmit(status)()
      }
```